### ISSUE TITLE
Custom order tables: fatal error when visiting Custom Data Stores

### ISSUE THREAD

**Issue by Chrystll** [woothemes/woocommerce#33189]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

**Installation**
WooCommerce 6.5.1 (migrating from 6.4)
WooCommerce Smooth Generator 1.0.4
Storefront 4.1.0
WordPress 6.0

I get a fatal error when going to WooCommerce > Settings > Advanced > Custom Data Stores.

```
Fatal error: Uncaught Error: Cannot access private const Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION in /home/christelle/Documents/sites/trunk-test/wp-content/plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php on line 158

Error: Cannot access private const Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION in /home/christelle/Documents/sites/trunk-test/wp-content/plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php on line 158

Call Stack
#	Time	Memory	Function	Location
1	0.0011	452376	{main}( )	.../admin.php:0
2	2.5379	61969160	do_action( )	.../admin.php:259
3	2.5379	61969536	WP_Hook->do_action( )	.../plugin.php:476
4	2.5379	61969536	WP_Hook->apply_filters( )	.../class-wp-hook.php:331
5	2.5379	61970664	WC_Admin_Menus->settings_page( )	.../class-wp-hook.php:307
6	2.5379	61970664	WC_Admin_Settings::output( )	.../class-wc-admin-menus.php:280
7	2.5383	61974920	include( '/home/christelle/Documents/sites/trunk-test/wp-content/plugins/woocommerce/includes/admin/views/html-admin-settings.php' )	.../class-wc-admin-settings.php:154
8	2.5400	61977328	do_action( )	.../html-admin-settings.php:40
9	2.5400	61977704	WP_Hook->do_action( )	.../plugin.php:476
10	2.5400	61977704	WP_Hook->apply_filters( )	.../class-wp-hook.php:331
11	2.5400	61978808	WC_Settings_Advanced->output( )	.../class-wp-hook.php:307
12	2.5400	61978808	WC_Settings_Advanced->output( )	.../class-wc-settings-advanced.php:448
13	2.5400	61978808	WC_Settings_Advanced->get_settings( )	.../class-wc-settings-page.php:211
14	2.5400	61978808	WC_Settings_Advanced->get_settings_for_section( )	.../class-wc-settings-page.php:98
15	2.5401	61978952	apply_filters( )	.../class-wc-settings-page.php:130
16	2.5401	61979352	WP_Hook->apply_filters( )	.../plugin.php:191
17	2.5401	61980832	Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController->Automattic\WooCommerce\Internal\DataStores\Orders\{closure}( )	.../class-wp-hook.php:307
18	2.5401	61980832	Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController->get_settings( )	.../CustomOrdersTableController.php:90
19	2.5663	62584128	Automattic\WooCommerce\Internal\DataStores\Orders\DataSynchronizer->get_sync_status( )	.../CustomOrdersTableController.php:307
20	2.5668	62579016	Automattic\WooCommerce\Internal\DataStores\Orders\DataSynchronizer->get_current_orders_pending_sync_count( )	.../DataSynchronizer.php:129
```

![screenshot-localhost-2022 05 25-11_21_49](https://user-images.githubusercontent.com/19824724/170230945-55e6cf8b-9445-45f2-b309-4721ec984de7.png)

### Expected behavior

I was expected to get access to the  "Custom Data Stores" (WooCommerce > Settings > Advanced).

### Actual behavior

I get the fatal error mentioned above.

### Steps to reproduce

This happens at step 4 of _Enable the feature_: https://developer.woocommerce.com/2022/05/16/call-for-early-testing-custom-order-table-migrations/.
Note that to enable the feature (step 1) I added the code snippet in the functions.php of my Storefront theme.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): http://localhost/trunk-test
Site address (URL): http://localhost/trunk-test
WC Version: 6.5.1
REST API Version: ✔ 6.5.1
WC Blocks Version: ✔ 7.4.3
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0
WP Multisite: –
WP Memory Limit: 4 GB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.29 (Unix) OpenSSL/1.0.2n PHP/7.2.3 mod_perl/2.0.8-dev Perl/v5.16.3
PHP Version: 7.2.3
PHP Post Max Size: 4 GB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.45.0
OpenSSL/1.0.2n

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.1.31-MariaDB
Max Upload Size: 4 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.4.0
WC Database Prefix: wp_
Total Database Size: 5.35MB
Database Data Size: 3.62MB
Database Index Size: 1.73MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.05MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.44MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.13MB + Index: 0.16MB + Engine InnoDB
wp_posts: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 11
page: 7
post: 2
product: 6
product_variation: 36
shop_order: 5

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (2) ###

WooCommerce Smooth Generator: by Automattic – 1.0.4
WooCommerce: by Automattic – 6.5.1

### Inactive Plugins (2) ###

Classic Editor: by WordPress Contributors – 1.6.2
Classic Menu Block: by Jonathan Harris – 0.1.5

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/
Checkout: #9 - /checkout/
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.1.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 69
Oldest: 2022-05-25 08:00:30 +0000
Newest: 2022-05-25 09:12:18 +0000

### Status report information ###

Generated at: 2022-05-25 09:25:00 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by swatipawarGS** [comment]
Hi @Chrystll,

Thank you for taking the time to report this bug, we really appreciate your help. We are able to reproduce the issue on our end using WooCommerce 6.5.1 version and WordPress 6.0 version. 

Please find below screenshot for reference :

![#33189](https://user-images.githubusercontent.com/94531721/170434347-e7a24d64-26f7-4e52-a272-2fb2bf6fadb5.jpg)

Priority of this issue will be assigned shortly after we’d go through a round of prioritization. Please note it may take a couple of days.

**Comment by tammullen** [comment]
Hi @Chrystll 

> This happens at step 4 of Enable the feature: https://developer.woocommerce.com/2022/05/16/call-for-early-testing-custom-order-table-migrations/.

Just noting from that post "The code for testing is currently not included in any official release."
I believe that is why you are seeing this error in 6.5.1 cc: @vedanshujain to confirm

You can use the build included in that post or a 6.6 pre release version such as 6.6 RC https://developer.woocommerce.com/2022/05/31/woocommerce-6-6-rc/

**Comment by Chrystll** [comment]
Hi @tammullen

Thank you. I don't get anymore the fatal error with the 6.6 RC2 version. That's great.
